src/actions/download.js

```diff
@@ -1,25 +1,13 @@
 import * as types from '../constants/actionTypes.js'
 
-// Set download state (download dialog is open)
-export const toggleDownloadDialog = (show) => ({
-    type: types.DOWNLOAD_DIALOG_TOGGLE,
-    payload: show,
+// Toggle download mode
+export const toggleDownloadMode = (payload) => ({
```

Comment:
I'm not sure it makes sense to call this action a "toggle" since you are sending a value with each call. If it is toggle, then the reducer should just reverse whatever value is currently set. That said, I think toggles can be untrustworthy, so maybe just rename this to `setDownloadMode`.

Reply:
I agree, renamed to `setDownloadMode` and `DOWNLOAD_MODE_SET`.